inst/include/Rcpp/Dimension.h

```diff
@@ -39,14 +39,14 @@ namespace Rcpp{
 	            dims = other.dims ;
 	        return *this ;
 	    }
-	    Dimension(const size_t& n1) : dims(1){
+	    Dimension(const int& n1) : dims(1){
```

Comment:
We are not that lighthearted about changing existing interfaces.  Right now I am inclined not to take the patch.   Lines 59 and 60 in `Dimension.h` look fine as does line 61 in `Matrix.h`.  I am not so sure about the rest.